### PR TITLE
Support dynamic `price_suffix` in in-app marketplace

### PR DESCRIPTION
**Pull request by andfinally**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds support for a `price_suffix` property in each product in the WooCommerce.com search and featured endpoints, which provide the data for the product cards in the in-app marketplace. This will allow us to list products whose fees are per month.

#### Note for GlobalStep testers

The required data changes for this PR haven't been implemented on WooCommerce.com yet, and the test below requires using the test data in a GitHub gist. For now, it's fine if we simply continue to see "per year" on the prices of products in the Marketplace, for example `$29.00 per year`, where the product is not free.

#### Test using dummy data

1. Check out this branch.
2. Edit `wp-content/plugins/woocommerce/plugins/woocommerce/includes/admin/class-wc-admin-addons.php:143` to point to this gist with test data: `https://gist.githubusercontent.com/andfinally/84015c40afc8dc628f2496c9c4faf0f8/raw/search.json`.

```php
$raw_extensions = wp_safe_remote_get(
	'https://gist.githubusercontent.com/andfinally/84015c40afc8dc628f2496c9c4faf0f8/raw/search.json',
	array( 'headers' => $headers )
);
```

3. Go to `wp-admin/admin.php?page=wc-addons&section=payment-gateways`.
4. Check that Authorize.Net has a price "per month".

image

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

Add - Support for dynamic price period in in-app marketplace product cards.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/admin/class-wc-admin-addons.php [hunk 1]
```diff
@@ -1284,7 +1296,13 @@ public static function render_product_card( $data, $block_type = null ) {
...
 									?>
 								</span>
-								<span class="price-suffix"><?php esc_html_e( 'per year', 'woocommerce' ); ?></span>
+								<span class="price-suffix"><?php
+									$price_suffix = __( 'per year', 'woocommerce' );
+									if ( ! empty( $mapped->price_suffix ) ) {
+										$price_suffix = $mapped->price_suffix;
+									}
+									echo esc_html( $price_suffix );;
+									?></span>
```
Comment (barryhughes) [inline plugins/woocommerce/includes/admin/class-wc-admin-addons.php]: Can we do:

```suggestion
								<span class="price-suffix">
									<?php
									$price_suffix = __( 'per year', 'woocommerce' );
									if ( ! empty( $mapped->price_suffix ) ) {
										$price_suffix = $mapped->price_suffix;
									}
									echo esc_html( $price_suffix );
									?>
								</span>
```

(This gets rid of the extra `;` on line 1304 and formatting it this way with the opening/closing PHP tags on their own lines silences the PHPCS warnings.)

**Review by barryhughes — CHANGES_REQUESTED** [review]
Just need a minor tweak to satisfy linting rules :+1:

File: plugins/woocommerce/includes/admin/class-wc-admin-addons.php (hunk 1 above)
Comment (andfinally) [reply plugins/woocommerce/includes/admin/class-wc-admin-addons.php]: Nice, thanks Barry.

**Review by barryhughes — APPROVED** [review]
LGTM!

(Triggered a re-run of a couple of checks, however the failures were likely unrelated to this change.)

**Comment by andfinally** [comment]
Great! Thanks for the review Barry.

**Pull request merged**